djangomodelimport/loaders.py

```diff
@@ -18,10 +29,10 @@ def __missing__(self, value):
 
         try:
             self[value] = inst = self.queryset.get(**params)
-        except self.model.DoesNotExist:
-            self[value] = inst = None  # Further warnings will be surpressed
+        except self.model.DoesNotExist as err:
+            self[value] = inst = err  # Further warnings will be re-raised
```

Comment:
No need to assign to `inst` if we're raising.
```suggestion
            self[value] = err  # Further warnings will be re-raised
```